synapseclient/extensions/curator/schema_generation.py

```diff
@@ -4263,7 +4312,13 @@ def _get_validation_rule_based_fields(
 
     Arguments:
         validation_rules: A list of input validation rules
+        explicit_is_array:
+          True: If the type is set explicitly with a list-type
+          False: If the type is set explicitly with a non list-type
+          None: If the type was nto set explicitly
```

Comment:
Based on the logic here: 
```
        if isinstance(column_type, AtomicColumnType):
            self.type = column_type
            explicit_is_array = False
        elif isinstance(column_type, ListColumnType):
            self.type = LIST_TYPE_DICT[column_type]
            explicit_is_array = True
        else:
            self.type = None
            explicit_is_array = False
```
I don't think `explicit_is_array` can be None?

Reply:
@linglp You are correct I meant to add this logic, but never did. I've added it, please take a look. Thanks!